### DISCUSSION TITLE
How to get the notes from a pattern

### DISCUSSION THREAD

**Discussion by curtistornade** [demberto/PyFLP#129] [Q&A] [answered]
Hello,
I'm struggling to get the different notes of a plugin inside a pattern. How do you create a pyflp.pattern.Note object ? If someone could give me an example of code it would be amazing.
Thank you !

**Comment by demberto** [comment, marked as answer]
All "model" classes in PyFLP are **proxy** classes. They are just a better representation of the underlying event data.

You cannot yet add notes, but you can remove or modify existing ones:
```py
import pyflp

project = pyflp.parse("/path/to/flp.flp")
for pattern in project.patterns:
    for note in pattern.notes:
        print(note)
        # You can do whatever modifications you need to here.
```

**Reply by curtistornade** [reply]
Ah perfect, thank you very much !
